src/CodeGen_Hexagon.cpp

```diff
@@ -1713,11 +1713,6 @@ void CodeGen_Hexagon::visit(const Call *op) {
             internal_assert(op->args.size() == 2);
             string instr = op->is_intrinsic(Call::shift_left) ? "halide.hexagon.shl" : "halide.hexagon.shr";
             Expr b = maybe_scalar(op->args[1]);
```

Comment:
If you remove this code here, then you'll have to handle signed variants of the shift amount in ```intrinsic_wrappers``` (https://github.com/halide/Halide/blob/master/src/CodeGen_Hexagon.cpp#L372) because we won't benefit anymore from the canonicalization done by ```lower_signed_shift```_xxx

Reply:
I think what needs to happen here is that we just change the wrappers to have signed RHSes, and cast unsigned RHSes to signed RHSes (might already happen automatically).

BTW, I think you meant to link to https://github.com/halide/Halide/blob/master/src/CodeGen_Hexagon.cpp#L604-L612.

We do need to be careful re: all of the patterns that involve shifts. It might be these basic shifts are the only ones with vector RHSes, and so this isn't actually tricky to handle.

I *think* it might only be v62 or later where the RHS can be signed. This isn't hard to support either, we basically just need to keep the old behavior if `get_target().has_feature(Target::v60)`. @pranavb-ca can you confirm? I read the HVX instruction docs but it was quite tricky to find exactly where this is supported. I'm pretty sure vasl, vasr, vlsr do after v62.

There's also this: https://github.com/halide/Halide/blob/master/src/runtime/hvx_64.ll#L288-L324 and https://github.com/halide/Halide/blob/master/src/runtime/hvx_128.ll#L289-L325. This is going to be a little annoying to handle. Given the nature of HVX intrinsics/type "safety" in LLVM, I think we can just make this all have a signed RHS, but only ever actually pass signed RHSes on v62 or later.

To summarize, I think what needs to happen is:
1. Change https://github.com/halide/Halide/blob/master/src/CodeGen_Hexagon.cpp#L604-L612 to have signed RHSes
2. Change https://github.com/halide/Halide/blob/master/src/runtime/hvx_64.ll#L288-L324 and https://github.com/halide/Halide/blob/master/src/runtime/hvx_128.ll#L289-L325 to have signed RHSes
3. Conditionalize the change you made here to only happen if `!get_target().has_feature(Target::v60)`
4. If necessary, we need to cast unsigned RHSes to signed RHSes here, but I think `call_intrin` already does this.

Reply:
Hi @dsharletg  - i like your suggestion. you are right, v60 doesn't have bidirectional shifts but every ISA after that does. So, yes, we should add a check for v60 and simply do 1, 2 above.

Reply:
This has been fixed now. Please take another look, thanks!